Events: Add placeholder to subqueries

wagner-certat opened the pull request:
see Intevation/intelmq-fody#70

bernhardreiter commented:
Hi @wagner-certat,
thanks for the pull request.

I've had a first look,
* To me it looks sensible to also add an example of a placeholder to the config file.
* An example for using `%` as post and prefix for substring search would be illustrative.
* The change introduces a dependency to `intelmq`  which wasn't there before. Probably fine, maybe listed in setup.py so this gets an automatic check. (Before in principle, someone could run the module without having a full intelmq installed, e.g. on a pure database machine.)

What do you think?

wagner-certat commented:
>     * To me it looks sensible to also add an example of a placeholder to the config file.

Good idea, done

>     * An example for using `%` as post and prefix for substring search would be illustrative.
Also added
>     * The change introduces a dependency to `intelmq`  which wasn't there before. Probably fine, maybe listed in setup.py so this gets an automatic check. (Before in principle, someone could run the module without having a full intelmq installed, e.g. on a pure database machine.)
It actually does only require the installed library, not the full (running) application. I didn't even check if the requirement is already there because I assumed it is. Thanks for the hint, I added it.

wagner-certat commented:
Is there interest in this feature?

bernhardreiter commented:
@wagner-certat  yes so far I think it is a useful feature. (We currently lack contracted time to properly review it.)

BTW: I wonder if the intelmq dependency could be made optional with little effort.

wagner-intevation commented:
> BTW: I wonder if the intelmq dependency could be made optional with little effort.

The patch uses two functions of the imported harmonization module:
- To generate the current time in the standardized format.
- To display a list of allowed classification types.

Both usages can be prevented by just copying the code from IntelMQ to fody.
We can also do both: try to import intelmq and use it's values if possible, and otherwise use the internal fallbacks.
